app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt

```diff
@@ -15,50 +16,53 @@ import kotlinx.coroutines.launch
 class SettingsActivity : AppCompatActivity(),
     ConnectDialog.ConnectDialogListener {
 
-    private var binding: ActivitySettingsBinding? = null
+    private lateinit var binding: ActivitySettingsBinding
 
     private var keyStore = KeyStoreHelper("connectToServer")
 
-    lateinit var connectionLiveData: ConnectionLiveData
+    private var isFragmentDurationValid: Boolean = false
 
-    lateinit var connectBtn : Button
+    lateinit var connectionLiveData: ConnectionLiveData
 
     private lateinit var sharedPreferences: SharedPreferences
     var pwdIV: String? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_settings)
+        sharedPreferences =
+            this.getSharedPreferences("com.camerasecuritysystem.client", Context.MODE_PRIVATE)
```

Comment:
Same with this one.

Reply:
No I will keep the shared preferences there, I like them to be initialized before the content gets set on the view.

Reply:
There is another reason for moving it down: In all the other activities, the view and bingings gets set or initialized first, and then the rest of the business logic. I would rather keep that consistent with the rest of the code.